scripts/ci/run-end-to-end-tests.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -ev
+
+handle_error() {
+    EXIT_CODE=$?
+    exit $EXIT_CODE
+}
+
+trap 'handle_error' ERR
```

Comment:
So I guess we're using separate scripts so that only this portion is covered by the `trap`...? I'm missing the context of what it achieves though I think...

Reply:
I'm not entirely sure what this `trap` achieves.
Maybe @t3chguy has more context as it's from his cherry picked commit. But this might be something that was carried over from the original script

Reply:
Yeah it's just copied over. Not sure if needed.

The point of splitting the script is to break down the gha log output so that it doesn't span 10k lines when only the last 20 matter

Reply:
IIRC, the only reason I used the traps is so in case of any error, the whole script will exit with a negative exit code and the CI environment knows it has to fail the tests.

Reply:
Looking at Michael's commit it appears that the `trap` was ported in both 
* `prepare-end-to-end-tests.sh`
* `run-end-to-end-tests.sh`

Meaning that the fallback originally implemented by Bruno will be applied if any of the scripts were to fail